app/Http/Controllers/Api/v1/PostRecoveryApiController.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Http\Controllers\Api\v1;
+
+use App\Models\Campaign;
+use App\Http\Resources\PostResource as Resource;
+use App\Http\Requests\RecoverPost as Request;
+use App\Services\RecoveryService;
+
+class PostRecoveryApiController extends ApiController
+{
+    protected RecoveryService $service;
+
+    public function __construct(RecoveryService $service)
+    {
+        $this->middleware('auth');
+
+        $this->service = $service;
+    }
+
+    /**
+     * @return \Illuminate\Http\Resources\Json\AnonymousResourceCollection
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function index(Campaign $campaign)
+    {
+        $this->authorize('access', $campaign);
```

Comment:
The permission should be `recover`, not `access`. It's also wrong in the other controller